src/app/plugins/slowmode.plugin.ts

```diff
@@ -0,0 +1,63 @@
+import { GuildChannel, TextChannel } from 'discord.js';
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IMessage } from '../../common/types';
+
+export class SlowModePlugin extends Plugin {
+  public name: string = 'SlowMode Plugin';
+  public description: string = "Restricts a user's access to specified channels";
+  public usage: string = 'slowmode <seconds to keep slowmode on> <slowmode setting> <channels...>';
+  public pluginAlias = ['slow'];
+  public permission: ChannelType = ChannelType.Staff;
+  public pluginChannelName: string = Constants.Channels.Staff.ModChat;
+
+  private _commandPattern: RegExp = /(\d+)\s+(\d+)\s+((?:<#(?:\d+)>\s*)+)/;
+  private _channelIDRegex: RegExp = /<#(\d+)>/g;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public validate(message: IMessage, args?: string[]) {
+    return !!args && this._commandPattern.test(args.join(' '));
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const match = args.join(' ').match(this._commandPattern);
+
+    // never happens, but make linter happy
+    if (!match) {
+      return;
+    }
+
+    const [_, expiration, slowmodeSetting, channels] = match;
+
+    const channel_objs =
+      channels
+        .match(this._channelIDRegex)
+        ?.map((c) => this.container.guildService.get().channels.get(c.replace(/\D/g, '')))
+        .filter((c) => c !== undefined) || [];
```

Comment:
random comment, you can do `.filter(Boolean)` here and becomes more readable (imo)